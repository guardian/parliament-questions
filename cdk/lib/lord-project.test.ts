import { App } from "aws-cdk-lib";
import { Template } from "aws-cdk-lib/assertions";
import { LordProject } from "./lord-project";

describe("The LordProject stack", () => {
  it("matches the snapshot", () => {
    const app = new App();
    const stack = new LordProject(app, "LordProject", { stack: "investigations", stage: "TEST" });
    const template = Template.fromStack(stack);
    expect(template.toJSON()).toMatchSnapshot();
  });
});
