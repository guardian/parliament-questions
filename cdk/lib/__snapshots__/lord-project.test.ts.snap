// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The LordProject stack matches the snapshot 1`] = `
{
  "Metadata": {
    "gu:cdk:constructs": [
      "GuDistributionBucketParameter",
      "GuScheduledLambda",
    ],
    "gu:cdk:version": "TEST",
  },
  "Parameters": {
    "DistributionBucketName": {
      "Default": "/account/services/artifact.bucket",
      "Description": "SSM parameter containing the S3 bucket name holding distribution artifacts",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "lordprojectDDCE0CB2": {
      "DependsOn": [
        "lordprojectServiceRoleDefaultPolicy6F577E43",
        "lordprojectServiceRole59EEDFD8",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Ref": "DistributionBucketName",
          },
          "S3Key": "investigations/TEST/lord-project/lord-project.zip",
        },
        "Environment": {
          "Variables": {
            "APP": "lord-project",
            "Bucket": "lord-project-dist",
            "LoggingLevel": "4",
            "STACK": "investigations",
            "STAGE": "TEST",
            "Stage": "TEST",
          },
        },
        "Handler": "dist/lambda/index.handler",
        "MemorySize": 512,
        "Role": {
          "Fn::GetAtt": [
            "lordprojectServiceRole59EEDFD8",
            "Arn",
          ],
        },
        "Runtime": "nodejs18.x",
        "Tags": [
          {
            "Key": "App",
            "Value": "lord-project",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "Stack",
            "Value": "investigations",
          },
          {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
        "Timeout": 45,
      },
      "Type": "AWS::Lambda::Function",
    },
    "lordprojectServiceRole59EEDFD8": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
        "Tags": [
          {
            "Key": "App",
            "Value": "lord-project",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "Stack",
            "Value": "investigations",
          },
          {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "lordprojectServiceRoleDefaultPolicy6F577E43": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      {
                        "Ref": "DistributionBucketName",
                      },
                    ],
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      {
                        "Ref": "DistributionBucketName",
                      },
                      "/investigations/TEST/lord-project/lord-project.zip",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": "ssm:GetParametersByPath",
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:aws:ssm:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":parameter/TEST/investigations/lord-project",
                  ],
                ],
              },
            },
            {
              "Action": [
                "ssm:GetParameters",
                "ssm:GetParameter",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:aws:ssm:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":parameter/TEST/investigations/lord-project/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "lordprojectServiceRoleDefaultPolicy6F577E43",
        "Roles": [
          {
            "Ref": "lordprojectServiceRole59EEDFD8",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "lordprojectlordprojectcron001020AllowEventRuleLordProjectlordprojectB499F0CFD2824A8D": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "lordprojectDDCE0CB2",
            "Arn",
          ],
        },
        "Principal": "events.amazonaws.com",
        "SourceArn": {
          "Fn::GetAtt": [
            "lordprojectlordprojectcron001020FFFFB711",
            "Arn",
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "lordprojectlordprojectcron001020FFFFB711": {
      "Properties": {
        "ScheduleExpression": "cron(00 10 ? * 2 *)",
        "State": "ENABLED",
        "Targets": [
          {
            "Arn": {
              "Fn::GetAtt": [
                "lordprojectDDCE0CB2",
                "Arn",
              ],
            },
            "Id": "Target0",
          },
        ],
      },
      "Type": "AWS::Events::Rule",
    },
  },
}
`;
